apps/web/src/pages/brand/tabs/LayoutEditor.tsx

```diff
@@ -1,38 +1,80 @@
 import { useForm, Controller } from 'react-hook-form';
 import { EmailCustomCodeEditor } from '../../../components/templates/email-editor/EmailCustomCodeEditor';
-import { Center, Grid } from '@mantine/core';
+import { Center, Grid, LoadingOverlay, useMantineTheme } from '@mantine/core';
 import { ArrowLeft } from '../../../design-system/icons';
 import { Button, Checkbox, colors, Input, Text } from '../../../design-system';
 import { useEnvController } from '../../../store/use-env-controller';
-import { successMessage } from '../../../utils/notifications';
+import { errorMessage, successMessage } from '../../../utils/notifications';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { createLayout, getLayoutById } from '../../../api/layouts';
+import { useEffect } from 'react';
 
-export function LayoutEditor({ id, goBack }: { id: string; goBack: () => void }) {
+export function LayoutEditor({
+  id = '',
+  editMode = false,
+  goBack,
+}: {
+  id?: string;
+  editMode?: boolean;
+  goBack: () => void;
+}) {
   const { readonly } = useEnvController();
+  const theme = useMantineTheme();
 
-  const { handleSubmit, watch, control } = useForm({
+  const { data: layout, isLoading: isLoadingLayout } = useQuery(['getLayout', id], () => getLayoutById(id), {
+    enabled: !!id,
+  });
+
+  const { handleSubmit, watch, control, setValue } = useForm({
     defaultValues: {
-      layout: '',
-      name: '',
+      content: layout?.content || '',
+      name: layout?.name || '',
       description: '',
-      isDefault: false,
+      isDefault: layout?.isDefault || false,
     },
   });
+  const { mutateAsync: createNewLayout, isLoading: isLoadingCreate } = useMutation(createLayout);
 
-  const content = watch('layout');
+  useEffect(() => {
+    if (layout) {
+      if (layout.content) {
+        setValue('content', layout?.content);
+      }
+      if (layout.name) {
+        setValue('name', layout?.name);
+      }
+      if (layout.isDefault != null) {
+        setValue('isDefault', layout?.isDefault);
+      }
+    }
+  }, [layout]);
 
   async function onUpdateLayout(data) {
     const updatePayload = {
       name: data.name,
-      content: data.layout,
+      content: data.content,
       isDefault: data.isDefault,
     };
 
-    successMessage('Layout updated!');
+    try {
+      await createNewLayout(data);
+    } catch (e: any) {
+      errorMessage(e.message || 'Un-expected error occurred');
+    }
+
+    successMessage(`Layout ${editMode ? 'Updated' : 'Created'}!`);
```

Comment:
I find weird that if we create a new layout and there is an error we send a success message.

Reply:
Oops. Good catch. Thanks